doc/tutorial/introduction-04.hpp

```diff
@@ -0,0 +1,113 @@
+#error This file is for documentation only - DO NOT INCLUDE
+/**
+
+@page intro-04 SIMD Data Structures
+
+@tableofcontents
+
+In the previous tutorial, we laid out how **EVE** can use SIMD-aware tuple to handle more complex
+cases. In this tutorial, we'll go over how we can design semantic-rich user-defined types to do the
+same.
+
+@snippet tutorial/intro-04.cpp  empty
+
+## Adapting UDT to SIMD processing
+Using tuples as a random bag of values returned from functions is sometime lackluster. Indeed, one
+may need to return data structure with a good level of semantic. User-defined types with named members
+is then preferred.
+
+Following this trend, we can rewrite our scalar `to_polar` function to return a proper structure.
+
+@snippet tutorial/intro-04.cpp scalar-udt
+
+The next thing we want is to be able to write `eve::wide<polar_coords>` and have everything we
+already put in place for handling tuple just works with user-defined type. **EVE** provides
+an opt-in system to do just that.
+
+To register `polar_coords` to be usable as a type inside eve::wide, two things are needed:
+  - adapt the structure to be [compatible with C++ structured bindings](https://en.cppreference.com/w/cpp/language/structured_binding).
+    This includes specializations for `std::tuple_size`, `std::tuple_element` and a `get` function.
+  - specialize eve::is_product_type to opt-in for product-type behavior
+
+@snippet tutorial/intro-04.cpp simd-udt-adapt
+
+The most interesting effect of this adaptation protocol is that you can adapt pre-existing
+user defined types without modifying their code, thus allowing for 3rd party types adaptations.
+
+We can then just rewrite `to_polar` to use this new UDT.
+
+@snippet tutorial/intro-04.cpp simd-udt-to_polar
+
+Note how the semantic improved by being able to explicitly states we return a SIMD register made
+of instance of `polar_coords`.
+
+Finally, the reliance on the structured binding protocol makes some standard trivial to adapt to
+SIMD processing, as only the specialization of eve::is_product_type is then required. This is the
+case for `std::array`for example.
+
+@code
+#include <array>
+#include <eve/wide.hpp>
+
+template<typename T, std::size_t N>
+struct eve::is_product_type<std::array<T,N>> : std::true_type
+{};
+@endcode
```

Comment:
Let's not suggest to do that? If we want to - we can, but I don't think our users should. Specialise for your types.
I'd drop this section about standard types all together.